xlsx_streaming/streaming.py

```diff
@@ -76,14 +77,15 @@ def stream_queryset_as_xlsx(
 
     return zipped_stream
 
-
 def serialize_queryset_by_batch(qs, serializer, batch_size):
-    start, last_batch = 0, []
-    while start == 0 or len(last_batch) == batch_size:
-        last_batch = list(qs[start:start + batch_size])  # force queryset evaluation
-        yield serializer(last_batch)
-        start += batch_size
-
+    qs_slices = chunks(qs, batch_size)
+    for batch in qs_slices:
+        yield serializer(list(batch))
+
+def chunks(iterable, size):
+    iterator = iter(iterable)
+    for first in iterator:
+        yield chain([first], islice(iterator, size - 1))
```

Comment:
From my understanding, this is not equivalent for django QuerySet to the previous implementation. Using `islice` all the data is fetched from the database at once while 'standard' slicing results instead of multiple SQL queries being actually triggered with `OFFSET xxx LIMIT yyy` arguments [1]. I am afraid the implementation merged here may cause performance issues when their are many results.
Using `.iterator()` [2] in the provided query would probably solve the issue, but I'm not sure if we should consider it to be the responsibility of this lib or of the calling code. In any way having a different `batch_size` for database fetching and serializing definitely sounds suboptimal.

[1] https://docs.djangoproject.com/en/2.2/topics/db/queries/#limiting-querysets
[2] https://docs.djangoproject.com/en/2.2/ref/models/querysets/#iterator